staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -284,9 +284,13 @@ func (p *jsonPatcher) applyJSPatch(versionedJS []byte) (patchedJS []byte, retErr
 	case types.JSONPatchType:
 		patchObj, err := jsonpatch.DecodePatch(p.patchJS)
 		if err != nil {
-			return nil, err
+			return nil, errors.NewBadRequest(err.Error())
```

Comment:
interpretPatchError can only deal with [mergepatch errors](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/mergepatch/errors.go#L26)

Reply:
Got it. 

* Can you remove the `interpretPatchError` at this line: https://github.com/kubernetes/kubernetes/blob/d472a5477705317706b09afc9589bb75496aeb02/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go#L268? Because it does nothing.

* Could you rename the `interpretPatchError` function to `interpretStrategicMergePatchError`, since that's what it handles.

I agree that we should fix upstream to make json-patch return typed errors. Then we can make `interpretPatchError` interprets json-patch errors.

Reply:
Done